services/placement/algo/sharded_helper.go

```diff
@@ -142,21 +142,27 @@ func (ph *placementHelper) GeneratePlacement() services.ServicePlacement {
 		}
 	}
 	return placement.NewPlacement().
-		SetInstances(ph.instances).
+		SetInstances(ph.getInstances()).
 		SetShards(ph.uniqueShards).
 		SetReplicaFactor(ph.rf).
 		SetIsSharded(true)
 }
 
-func (ph placementHelper) PlaceShards(shards []shard.Shard, from services.PlacementInstance) error {
+func (ph *placementHelper) PlaceShards(shards []shard.Shard, from services.PlacementInstance) error {
 	shardSet := getShardMap(shards)
 	if from != nil {
+		// NB(cw) when removing an adding instance that has not finished bootstrapping its
+		// Initializing shards, prefer to return those Initializing shards back to the leaving instance
+		// to reduce some bootstrapping work in the cluster.
+		if err := ph.returnInitializingShardsToSource(shardSet, from); err != nil {
```

Comment:
Will this method modify `shardSet` or `from` in any way? If so, would it be okay if it returns an error and `shardSet` and `from` has been changed?